pyfftw/interfaces/scipy_fft.py

```diff
@@ -0,0 +1,342 @@
+
+
+@_implements(_fft.rfft)
+def rfft(x, n=None, axis=-1, norm=None, overwrite_x=False,
+        planner_effort=None, threads=None,
+        auto_align_input=True, auto_contiguous=True):
+    '''Perform a 1D real FFT.
+
+    The first five arguments are as per :func:`scipy.fft.rfft`;
+    the rest of the arguments are documented
+    in the :ref:`additional argument docs<interfaces_additional_args>`.
+    '''
+    x = np.asanyarray(x)
+    if not np.isrealobj(x):
+        raise TypeError('x must be a real sequence')
+
+    return numpy_fft.rfft(x, n, axis, norm, overwrite_x, planner_effort,
+                          threads, auto_align_input, auto_contiguous)
+
+@_implements(_fft.irfft)
+def irfft(x, n=None, axis=-1, norm=None, overwrite_x=False,
+        planner_effort=None, threads=None,
+        auto_align_input=True, auto_contiguous=True):
+    '''Perform a 1D real inverse FFT.
+
+    The first five arguments are as per :func:`scipy.fft.irfft`;
+    the rest of the arguments are documented
+    in the :ref:`additional argument docs<interfaces_additional_args>`.
+    '''
+    return numpy_fft.irfft(x, n, axis, norm, overwrite_x, planner_effort,
+                           threads, auto_align_input, auto_contiguous)
+
+
+@_implements(_fft.rfft2)
+def rfft2(x, shape=None, axes=(-2,-1), norm=None, overwrite_x=False,
+        planner_effort=None, threads=None,
+        auto_align_input=True, auto_contiguous=True):
+    '''Perform a 2D real FFT.
+
+    The first five arguments are as per :func:`scipy.fft.rfft2`;
+    the rest of the arguments are documented
+    in the :ref:`additional argument docs<interfaces_additional_args>`.
+    '''
+    x = np.asanyarray(x)
+    if not np.isrealobj(x):
+        raise TypeError('x must be a real sequence')
+
+    return numpy_fft.rfft2(x, shape, axes, norm, overwrite_x,
+            planner_effort, threads, auto_align_input, auto_contiguous)
+
+
+@_implements(_fft.irfft2)
+def irfft2(x, shape=None, axes=(-2,-1), norm=None, overwrite_x=False,
+        planner_effort=None, threads=None,
+        auto_align_input=True, auto_contiguous=True):
+    '''Perform a 2D real inverse FFT.
+
+    The first five arguments are as per :func:`scipy.fft.irfft2`;
+    the rest of the arguments are documented in the
+    :ref:`additional argument docs <interfaces_additional_args>`.
+    '''
+    return numpy_fft.irfft2(x, shape, axes, norm, overwrite_x,
+            planner_effort, threads, auto_align_input, auto_contiguous)
+
+
+@_implements(_fft.rfftn)
+def rfftn(x, shape=None, axes=None, norm=None, overwrite_x=False,
+        planner_effort=None, threads=None,
+        auto_align_input=True, auto_contiguous=True):
+    '''Perform an n-D real FFT.
+
+    The first five arguments are as per :func:`scipy.fft.rfftn`;
+    the rest of the arguments are documented
+    in the :ref:`additional argument docs<interfaces_additional_args>`.
+    '''
+    x = np.asanyarray(x)
+    if not np.isrealobj(x):
```

Comment:
Using `if np.iscomplexobj(x)` should have slightly less overhead
 (`isrealobj` calls `iscomplexobj` internally)

Alternatively, `if x.dtype.kind == 'c':` should be even faster than calling `np.iscomplexobj`.